src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java

```diff
@@ -131,14 +137,26 @@ public void connect(MessageContext messageContext) {
             org.apache.kafka.common.header.Headers headers = getDynamicParameters(messageContext);
             publishMessage(messageContext, topic, key, partitionNo, value, headers);
 
-        } catch (AxisFault axisFault) {
-            handleException("Kafka producer connector " +
-                                    ": Error sending the message to broker lists", axisFault, messageContext);
-        } catch (SerializationException e) {
-            handleException("Error sending the Avro message to broker", e, messageContext);
+        } catch (Exception e) {
+            handleError(messageContext, e, Utils.getErrorCode(e),
+                    "Kafka producer connector : Error sending the message to broker");
         }
     }
 
+    /**
+     * Sets error to context and handle.
+     *
+     * @param msgCtx      Message Context to set info
+     * @param e           Exception associated
+     * @param error       Error code
+     * @param errorDetail Error detail
+     */
+    private void handleError(MessageContext msgCtx, Exception e, Error error, String errorDetail) {
+
+        Utils.setError(msgCtx, e, error);
```

Comment:
Shall we use setErrorPropertiesToMessage and remove the setError method? WDYT?

Reply:
Fixed with ba1167b